### PR TITLE
Sort protocols by workflow order

### PR DESCRIPTION
**Pull request by ckostiw**
Display protocols in the labwork overview page in a sensible order - the order in which they normally appear in workflows.

The idea is to look at each protocol and see which protocols it should follow and which it should precede based on where it appears in the workflows.

This gets done once, the first time the labwork is fetched from the backend, after the app is initialized and all the workflow definitions have been loaded.

I also tweaked the labwork-info endpoint in sample_next_step to avoid returning info about steps that are not used in any workflow - I was getting an Infinium protocol that is not used in any workflow and always had zero samples.

### CODE REVIEW

File: backend/fms_core/viewsets/sample_next_step.py [hunk 1]
```diff
@@ -170,6 +170,12 @@ def labwork_info(self, request, *args, **kwargs):
             if protocol.id not in sample_next_step_by_protocol.keys():
 
                 protocolSteps = Step.objects.filter(protocol=protocol)
+
+                # Make sure that at least one workflow uses one of the steps for this protocol, since
+                # labwork doesn't need protocols that are not used by any workflow (Infinium...)
+                workflows = Workflow.objects.filter(steps__in=protocolSteps)
```
Comment (UlysseFG) [inline backend/fms_core/viewsets/sample_next_step.py:176]: `if Workflow.objects.filter(steps__in=protocolSteps).exists():` should be lighter and clearer.

File: frontend/src/modules/labwork/services.ts [hunk 2]
```diff
@@ -154,4 +173,99 @@ export function findChangedStepsInSummary(oldSummary: LabworkSummary, newSummary
...
+				const distanceA = entriesA.reduce((acc, entry) => {
+					return acc + (entry.stepOrder / entry.numSteps)
+				}, 0) / entriesA.length
+				const distanceB = entriesB.reduce((acc, entry) => {
+					return acc + (entry.stepOrder / entry.numSteps)
+				}, 0) / entriesB.length
+
+				return distanceA - distanceB
+			} else {
+				return numAfter - numBefore
```
Comment (UlysseFG) [inline frontend/src/modules/labwork/services.ts:261]: It's likely going to give wonky results as we are not using a common metric for all steps... Both metrics are not on the same scale... This seem to be working for now but it's likely to need rework down the line. Let's roll with it, but ordering usually use only a single criterion (or a weighted combination).

File: backend/fms_core/viewsets/sample_next_step.py (hunk 1 above)
Comment (ckostiw) [reply backend/fms_core/viewsets/sample_next_step.py]: Excellent - fixed. 👍

File: frontend/src/modules/labwork/services.ts (hunk 2 above)
Comment (ckostiw) [reply frontend/src/modules/labwork/services.ts:261]: Yes, I'm sure that there is a better algorithm - this is a bit hacky, I agree, but it got a decent result for now.

**Review by UlysseFG — APPROVED** [review]

**Pull request merged**